pkg/kubernetes/client/diff_test.go

```diff
@@ -9,48 +9,70 @@ import (
 
 func TestSeparateMissingNamespace(t *testing.T) {
 	cases := []struct {
-		name string
-		td   nsTd
+		name      string
+		td        nsTd
+		namespace string // namespace as defined in spec.json
 
 		missing bool
 	}{
 		// default should always exist
 		{
-			name: "default",
+			name:      "default",
+			namespace: "default",
 			td: newNsTd(func(m manifest.Metadata) {
 				m["namespace"] = "default"
 			}, []string{}),
 			missing: false,
 		},
-		// implcit default (not specfiying an ns at all) also
+		// implicit default (not specfiying an ns at all) also
 		{
-			name: "implicit-default",
+			name:      "implicit-default",
+			namespace: "default",
 			td: newNsTd(func(m manifest.Metadata) {
 				delete(m, "namespace")
 			}, []string{}),
 			missing: false,
 		},
 		// custom ns that exists
 		{
-			name: "custom-ns",
+			name:      "custom-ns",
+			namespace: "custom",
 			td: newNsTd(func(m manifest.Metadata) {
 				m["namespace"] = "custom"
 			}, []string{"custom"}),
 			missing: false,
 		},
 		// custom ns that does not exist
 		{
-			name: "missing-ns",
+			name:      "missing-ns",
+			namespace: "missing",
 			td: newNsTd(func(m manifest.Metadata) {
 				m["namespace"] = "missing"
 			}, []string{}),
 			missing: true,
 		},
+		// an explicitly created namespace is missing
+		{
+			name:      "explicit-namespace-missing",
+			namespace: "other-namespace",
+			td: newNsTd(func(m manifest.Metadata) {
+				m["kind"] = "Namespace"
+				m["name"] = "explicit-namespace"
+			}, []string{}),
+			missing: true,
+		},
+		// an environment's namespace is missing and resources have no ns defined
+		{
+			name:      "env-namespace-missing",
+			namespace: "env-namespace",
+			td:        newNsTd(func(m manifest.Metadata) {}, []string{}),
+			missing:   true,
+		},
```

Comment:
I think this case is wrong, because it can't reliably detect the difference between:

- a namespaced resource with no namespace set (implicit default)
- a non-namespaced resource

So now, all cluster wide resources will always be marked as missing